xarray/core/dataset.py

```diff
@@ -3558,6 +3557,13 @@ def drop(  # noqa: F811
                 ds = ds._drop_labels(labels, dim, errors=errors)
             return ds
         elif dim is None:
+            if isinstance(labels, DataArrayCoordinates):
```

Comment:
I think this should probably be even more generic:
```suggestion
            if is_dict_like(labels) and not isinstance(labels, dict):
```
and probably moved up to ensure it gets triggered in all situations, not just if `dim is None`.

Reply:
Sure, makes sense. I tried to write an error message that captured the generic case as well.